### ISSUE TITLE
Dates not being interpreted by Tableau

### ISSUE THREAD

**Issue by CamDL** [innobi/pantab#66]
**Describe the bug**
After the 0.2.0 release, datetime data types written to the .hyper file are not being interpreted properly by Tableau Desktop. Loading data from the .hyper file to a new pandas dataframe displays the correct date/time information. This only affects dates that have been parsed with pandas and does not affect dates stored as an object.

Reverting back to pantab version 0.1.1 resolved the issue.

**To Reproduce**

```python
import pandas as pd
import datetime
import pantab
  
url = 'http://bit.ly/uforeports'
  
# read csv file 
df = pd.read_csv(url, low_memory=False, parse_dates=['Time'])

# Time should have a data type of datetime
print(df.info())

# write the dataframe to a hyper file
pantab.frame_to_hyper(df, 'UFO.hyper', table="UFO")

# read the hyper to a new dataframe
df2 = pantab.frame_from_hyper('UFO.hyper',table="UFO")

# display the top rows - dates/times come back properly
print(df2.head())>
```
Next, open the created .hyper file with Tableau Desktop. Describing the Time column shows "Error" for the domain.

**Expected behavior**
Expected behavior is that the Time column is properly interpreted by Tableau as a series of Nulls and Dates/Times.

**Screenshots**
![image](https://user-images.githubusercontent.com/4852986/71533739-91a06680-28c8-11ea-922f-6bf324a73256.png)

**Desktop (please complete the following information):**
 - OS: Windows 10 Enterprise
 - Python: 3.7
 - Tableau: Desktop Professional 2019.2
 - Hyper API: 0.0.9273
 - Pantab: 0.2.2

**Comment by WillAyd** [comment]
Is it just NA timestamps that are problematic?

**Comment by WillAyd** [comment]
Hmm I can't seem to reproduce this:

<img width="733" alt="image" src="https://user-images.githubusercontent.com/609873/71537207-a12ea800-28e6-11ea-96f5-f007936c424a.png">

I am on macOS with Python 37, pantab 0.2.2 and using Hyper API 8707. Tableau Desktop is 2019.4

Do you have any way of finding the problematic rows on your end?

**Comment by WillAyd** [comment]
Upgraded to Hyper API 0.0.9273 and couldn't reproduce either

**Comment by CamDL** [comment]
**Opening Existing Extract on Mac**
Opening the file that I had created on my PC still brought in Error values on my Mac.
![image](https://user-images.githubusercontent.com/4852986/71538788-5d946800-28ff-11ea-94a9-c2fe7ce7f4ab.png)

**Creating New Extract on Mac**
Running the exact same script as before, but from my Mac with pantab 0.2.2 yielded no errors when opening the extract in Tableau. This same Hyper file can be opened and viewed on my PC with no problem.
![image](https://user-images.githubusercontent.com/4852986/71538798-8b79ac80-28ff-11ea-8e4b-c8c7fdce9656.png)

It appears to be either specific to my machine or to Windows. I know that the Hyper API for Windows requires the installation of a Visual C++ Runtime Library, so it's possible that the issue lies with the Hyper API itself. I'll test running the code on another PC tomorrow.

**Comment by CamDL** [comment]
> Is it just NA timestamps that are problematic?

Tried running this on my PC dropping all NA values from the dataframe before writing, but still received the same Error in Tableau. Reading the values back to a new dataframe with pantab yielded the correct dates.

**Comment by WillAyd** [comment]
Interesting finds - keep me posted!

**Comment by CamDL** [comment]
Dug into it a little more by accessing the .hyper file directly with the Hyper API. It looks like it's encoding the date incorrectly. My guess is that this has to do with Windows epochs vs Unix epochs, but I'm not very familiar with C.

Sample row from PC is below:
['Belton', 'RED', 'SPHERE', 'SC', Timestamp(-4674, 6, 11, 12, 19, 11, 385600)]

Sample row from Mac is below:
['Belton', 'RED', 'OVAL', 'MO', Timestamp(1999, 11, 23, 18, 10, 0, 0)]

**Comment by CamDL** [comment]
[UFO-Hypers.zip](https://github.com/innobi/pantab/files/4006753/UFO-Hypers.zip)
Hyper files attached here.

**Comment by WillAyd** [comment]
So you see this as having started in 0.2.0 right? If I had to guess it may have to do with the definition of the MICROSECONDS_PER_DAY macro definition which was moved and explicitly cast to uint64_t here

https://github.com/innobi/pantab/commit/0dc969aa53add3404d2731786c2290eceb6ae996

I’m away from a computer until the new year but would certainly welcome a PR to fix if you see what there may have caused the regression. Timestamps are int64_t values representing nanoseconds since the Unix epoch, so perhaps Windows doesn’t like the signed / unsigned arithmetic and that macro needs to be int64_t

An easy way to check what is going on is to add print statements wherever the value appears like “printf(“%” PRint64 “\n”, <variable>) and see what shows in the terminal. 

@cedricyau
